### ISSUE TITLE
NSPropertyListSerialization not handling deserialisation of NSDate objects correctly.

### ISSUE THREAD

**Issue by rupertdaniel** [Microsoft/WinObjC#514]
NSPropertyListSerialization is not deserialising dates correctly. 

For example, when deserialising  the following plist...

```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple Computer//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
	<dict>
		<key>Date Modified</key><date>2014-12-15T19:48:38Z</date>
		<key>Date Added</key><date>2016-04-29T14:51:17Z</date>
	</dict>
</plist>
```

NSPropertyListSerialization will create the following NSDate objects... 

```
{
    "Date Added" = 2001-01-01 00:33:36 +0000
    "Date Modified" = 2001-01-01 00:33:34 +0000
}
```

**Comment by rajsesh-msft** [comment]
NSXMLPropertyList.mm has this code:

```c
static id propertyListFromElement(xml_node<>* elem) {
...
`    } else if (strncmp(elem->name(), "date", elem->name_size()) == 0) {
        Str numStr(elem->value(), elem->value_size());
        return [NSDate dateWithTimeIntervalSinceReferenceDate:atof(numStr.cstr())];
> there-in lies the rub. The value is string representation. Should use NSDateFormatter dateFromString

    } else {
```

**Comment by msft-Jeyaram** [comment]
@rupertdaniel @rajsesh-msft  the fix is available as per 2e5dc0c6fa44c68040ec3b90ee1cda3053a690b2
